sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/MetricsAdvisorClient.cs

```diff
@@ -1137,27 +1119,7 @@ public virtual AsyncPageable<IncidentRootCause> GetIncidentRootCausesAsync(Anoma
         {
             Argument.AssertNotNull(incident, nameof(incident));
 
-            Guid detectionConfigurationGuid = new Guid(incident.DetectionConfigurationId);
-            string incidentId = incident.Id;
-
-            async Task<Page<IncidentRootCause>> FirstPageFunc(int? pageSizeHint)
-            {
-                using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(MetricsAdvisorClient)}.{nameof(GetIncidentRootCauses)}");
-                scope.Start();
-
-                try
-                {
-                    Response<RootCauseList> response = await _serviceRestClient.GetRootCauseOfIncidentByAnomalyDetectionConfigurationAsync(detectionConfigurationGuid, incidentId, cancellationToken).ConfigureAwait(false);
-                    return Page.FromValues(response.Value.Value, null, response.GetRawResponse());
-                }
-                catch (Exception e)
-                {
-                    scope.Failed(e);
-                    throw;
-                }
-            }
-
-            return PageableHelpers.CreateAsyncEnumerable(FirstPageFunc, null);
+            return GetIncidentRootCausesAsync(incident.DetectionConfigurationId, incident.Id, cancellationToken);
```

Comment:
Removing almost the whole method as it's just a copy of another one. Decided to just pass the right arguments to avoid repeating code.